common/ledger/blkstorage/blockfile_mgr.go

```diff
@@ -323,31 +363,42 @@ func (mgr *blockfileMgr) addBlock(block *common.Block) error {
 
 func (mgr *blockfileMgr) syncIndex() error {
 	var lastBlockIndexed uint64
-	var indexEmpty bool
+	lastBlockIndexMarkerPresent := true
 	var err error
 	//from the database, get the last block that was indexed
 	if lastBlockIndexed, err = mgr.index.getLastBlockIndexed(); err != nil {
-		if err != errIndexEmpty {
+		if err != errIndexCheckpointKeyNotPresent {
 			return err
 		}
-		indexEmpty = true
+		lastBlockIndexMarkerPresent = false
+	}
+
+	if mgr.bootstrappedFromSnapshot() && !lastBlockIndexMarkerPresent {
+		// This condition can happen only if there was a peer crash or failure during
+		// bootstrapping the ledger from a snapshot or the index is dropped/corrupted afterward
+		return errors.Errorf(
+			"cannot sync index with block files. blockstore is bootstrapped from a snapshot and first available block=[%d]",
+			mgr.firstPossibleBlockNumberInBlockFiles(),
+		)
+	}
```

Comment:
Clarification question.

The syncIndex() can be called in the following scenarios:

1. Brand new chain -- _line 387 would return_ 
2. Opening an old chain after a failure (non-snapshot based chain) -- _regular check and sync if needed_
3. Opening a just bootstrapped chain -- _line 387 would return_ 
4. Opening a old bootstrapped chain -- _same as the 2_
5. Opening an incompletely bootstrapped chain -- _we return an error. When the consumer does not take care of the failure/recovery properly, this would be helpful_ 

Hope, I covered all scenarios.

Reply:
Question is wether we have these 5 scenarios only or am I missing something? (just double checking my understanding